File: Arrays/Algorithms/Sliding_Window/Variable_Size/smallest_subarray_with_sum_greater_than_k.py
def smallest_subarray_with_sum_greater_than_k(arr, k):
	n = len(arr)
	l = s = 0
	res = n + 1
	for r in range(n):
		s += arr[r]
		while s > k:
			res = min(res, r - l + 1)
			s -= arr[l]
			l += 1
	return 0 if res == n + 1 else res

if __name__ == "__main__":
	n, k = map(int, input().split())
	arr = list(map(int, input().split()))
	print(smallest_subarray_with_sum_greater_than_k(arr, k))

File: Arrays/Algorithms/Sliding_Window/Variable_Size/smallest_subarray_with_sum_greater_than_k.cpp
#include <bits/stdc++.h>
using namespace std;

int smallestSubarrayWithSumGreaterThanK(const vector<int>& arr, int k) {
	int n = arr.size(), l = 0, sum = 0, res = n + 1;
	for (int r = 0; r < n; r++) {
		sum += arr[r];
		while (sum > k) {
			res = min(res, r - l + 1);
			sum -= arr[l++];
		}
	}
	return res == n + 1 ? 0 : res;
}

int main() {
	int n, k;
	cin >> n >> k;
	vector<int> arr(n);
	for (int i = 0; i < n; i++) cin >> arr[i];
	cout << smallestSubarrayWithSumGreaterThanK(arr, k) << endl;
	return 0;
}
